auth/authpb/auth.proto

```diff
@@ -16,7 +16,21 @@ message User {
   int64 tombstone = 3;
 }
 
+// Permission is a single entity
+message Permission {
+  bytes path = 1;
+
+  enum PermissionType {
+    READ = 0;
+    WRITE = 1;
+  }
+  int32 permissionBitmap = 2;
+}
+
 // Role is a single entry in the bucket authRoles
 message Role {
-  bytes name = 2;
+  bytes name = 1;
+
+  repeated Permission pathPermission = 2;
```

Comment:
> BTW, what is the main motivation of not using the v2's file system like model and using the flat key model in v3? Supporting range query? Just curious.

Making the abstraction simpler and allowing more flexible key space operation. Now it is just a binary key without any restriction.